apis/openstack-nova/src/main/java/org/jclouds/openstack/nova/v2_0/compute/functions/CleanupServer.java

```diff
@@ -95,6 +101,7 @@ public Boolean apply(String id) {
       }
 
       boolean serverDeleted = novaApi.getServerApi(regionAndId.getRegion()).delete(regionAndId.getId());
+      checkState(nodeTerminatedPredicate.apply(id), "server was not destroyed in the configured timeout");
```

Comment:
Thinking about it more, I guess `destroyNode` should not invoke `CleanUpServer` but simply
```
      boolean serverDeleted = novaApi.getServerApi(regionAndId.getRegion()).delete(regionAndId.getId());
      checkState(nodeTerminatedPredicate.apply(id), "server was not destroyed in the configured timeout");
```
as `CleanUpServer` will be invoked by https://github.com/jclouds/jclouds/blob/master/apis/openstack-nova/src/main/java/org/jclouds/openstack/nova/v2_0/compute/NovaComputeService.java#L99 anyways. I'll test it, meanwhile wdyt?